### PR TITLE
sigi: 3.6.1 -> 3.6.3

### PR DESCRIPTION
**Pull request by booniepepper**
## Description of changes

Changes are minor updates and additions:

https://github.com/sigi-cli/sigi/compare/v3.6.1...v3.6.3

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/applications/misc/sigi/default.nix [hunk 1]
```diff
@@ -2,14 +2,14 @@
...
+  version = "3.6.3";
 
   src = fetchCrate {
     inherit pname version;
-    sha256 = "sha256-UL4V/5XvqaqO4R2ievw379D/rzHf/ITgvG3BcSbMeTQ=";
+    sha256 = "sha256-JGQ9UbkS3Q1ohy6vtiUlPijuffH4Gb99cZCKreGqE/U=";
   };
 
-  cargoSha256 = "sha256-wzTUK4AvJmBK7LX7CLCAeAXLDxMJA/3qs/KT1+pMaoI=";
+  cargoSha256 = "sha256-W/ekk4tsYxG7FXzJW5i0Ii7nLgDHCSCjO3couN+/sMk=";
```
Comment (kirillrdy) [inline pkgs/applications/misc/sigi/default.nix:12]: https://nixos.org/manual/nixpkgs/stable/#rust

consider changing to `cargoHash` and `hash`

`buildRustPackage requires either the cargoSha256 or the cargoHash attribute which is computed over all crate sources of this package. cargoHash256 is used for traditional Nix SHA-256 hashes, such as the one in the example above. cargoHash should instead be used for [SRI](https://www.w3.org/TR/SRI/) hashes`

Comment (booniepepper) [reply pkgs/applications/misc/sigi/default.nix]: I'm not quite sure I understand the benefits or preference from the docs... but I'm all for trying it out. Updated with this

Comment (kirillrdy) [reply pkgs/applications/misc/sigi/default.nix]: old attributes only supported single  of hash algorith eg sha256, which means when when calling overrideAttrs, you had to know what type of hash each derivation used

with  `hash` attribute, you can use SRI hash notation ( which you already using ) and then you can use different type of hash algorithm with same `hash` attribute

Comment (booniepepper) [reply pkgs/applications/misc/sigi/default.nix]: Ahh, great!

**Comment by kirillrdy** [comment]
@booniepepper can you please include link to changelog and/or diff in the commit message and PR

and I guess we can close https://github.com/NixOS/nixpkgs/pull/261192

**Comment by booniepepper** [comment]
@kirillrdy updated the commit & PR with diffs. It's mostly docs and dependency updates. There's one non-breaking alias added for a command in the program's interactive mode

Also I closed https://github.com/NixOS/nixpkgs/pull/261192 since this overrides it

Thanks again for looking!

**Review by kirillrdy — APPROVED** [review]
lets wait for CI to pass

**Pull request merged**